financepayment/controllers/front/response.php

```
@@ -640,4 +632,146 @@ protected function createSignature($payload, $secret)
 
         return $signature;
     }
+
+    /**
+     * checks the integrity and validity of the webhook payload
+     *
+     * @param string $input the raw string of the body
+     * @return object the json decoded webhook payload object
+     * @throws WebhookException if json could not be decoded
+     * @throws WebhookException if secret set but no HMAC received in the header
+     * @throws WebhookException if divido HMAC does not match HMAC generated with config shared secret
+     * @throws WebhookException if a required element of the payload is missing
+     */
+    protected function validateWebhook($input){
+        $data  = json_decode($input);
+        if(!$data){
+            throw new WebhookException("Could not decode json payload of request", self::HTTP_RESPONSE_CODE_BAD_REQUEST);
+        }
```

I'm always a fan of using the `JSON_THROW_ON_ERROR` flag (available since php 7.3) to throw a [JsonException](https://www.php.net/manual/en/class.jsonexception.php), to give more context as to why decoding failed. 

You could then explicitly catch that and throw your WebhookException containing that context, or just let the JsonException bubble up